File: AN_2/SEM_1/MAP/lab2/src/container/QueueContainer.java
package container;

import model.Task;

public class QueueContainer extends AbstractContainer
{
    public QueueContainer()
    {
        taskuri=new Task[10];
        size=0;
    }

    @Override
    public Task remove()
    {
        if(!isEmpty())
        {
            Task task=taskuri[0];
            for(int i=0;i<size-1;i++)
                taskuri[i]=taskuri[i+1];
            taskuri[--size]=null;
            return task;
        }
        return null;
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Container/QueueContainer.cs
using Model;

namespace Container
{
    public class QueueContainer : AbstractContainer
    {
        public QueueContainer()
        {
            taskuri = new Tasks[10];
            size = 0;
        }

        public override Tasks Remove()
        {
            if (!IsEmpty())
            {
                Tasks task = taskuri[0];
                for (int i = 0; i < size - 1; i++)
                    taskuri[i] = taskuri[i + 1];
                taskuri[--size] = null;
                return task;
            }
            return null;
        }
    }
}
